test/src/test/java/org/corfudb/integration/MergeTestListener.java

```diff
@@ -0,0 +1,228 @@
+ *      and apply the keys to merged table
+ *  (5) Process updates for log entry sync
+ */
+@Slf4j
+public class MergeTestListener extends LogReplicationListener{
+
+
+    public CorfuRuntime runtime;
+
+
+    private CorfuStore corfuStoreSink;
+
+    private final String namespace = "LR-Test";
+
+    String mergedTableName = "MERGED_TABLE";
+
+
+    // Updates received through streaming
+    @Getter
+    public static LinkedList<CorfuStreamEntry> updates = new LinkedList<>();
+
+    public Set<Message> deletedKeys = new HashSet<>();
+
+    public Map<Message, CorfuStoreEntry> addedKeys = new HashMap<>();
+
+    public Set<Message> existingKeysInMergedTable = new HashSet<>();
+
+    public static Table<Sample.StringKey, SampleSchema.SampleMergedTable, Message> mergedTable;
+
+    List<Table<Sample.StringKey, SampleSchema.SampleGroupMsgA, Message>> tableListForGroupA;
+
+    List<Table<Sample.StringKey, SampleSchema.SampleGroupMsgA, Message>> tableListForLocalTable;
+
+    public static List<CorfuStoreEntry<Sample.StringKey, SampleSchema.SampleGroupMsgA, Message>> entries
+            = new ArrayList<>();
+
+    MergeTestListener(CorfuStore corfuStoreSink, String namespace, List<Table<Sample.StringKey, SampleSchema.SampleGroupMsgA, Message>>
+                      tableListForGroupA, List<Table<Sample.StringKey, SampleSchema.SampleGroupMsgA, Message>> tableListForLocalTable
+    ) throws InvocationTargetException, NoSuchMethodException, IllegalAccessException {
+        super(corfuStoreSink, namespace);
+        this.corfuStoreSink = corfuStoreSink;
+        openMergedTable();
+        this.tableListForGroupA = tableListForGroupA;
+        this.tableListForLocalTable = tableListForLocalTable;
+    }
+
+    @Override
+    public void onError(Throwable throwable) {
+        log.error("Error on listener", throwable);
+        fail("onError for LRTestListener: " + throwable.toString());
+    }
+
+    /**
+     * Invoke when snapshot sync starts
+     */
+    @Override
+    protected void onSnapshotSyncStart() {
+        log.info("Snapshot sync started");
+    }
+
+    /**
+     * Apply addedKeys and deletedKeys to merged table, add keys to
+     * existingKeysInMergedTable from addedKeys and deletedKeys.
+     */
+    @Override
+    protected void onSnapshotSyncComplete() {
+        try (TxnContext tx = corfuStoreSink.txn(NAMESPACE)){
+            existingKeysInMergedTable.addAll(addedKeys.keySet());
+            addedKeys.forEach((addedKey, e) -> {
+                tx.putRecord(mergedTable, (Sample.StringKey) addedKey,
+                        SampleSchema.SampleMergedTable.newBuilder().setPayload(e.getPayload().toString()).build(), e.getMetadata());
+            });
+
+            existingKeysInMergedTable.removeAll(deletedKeys);
+            deletedKeys.forEach(deletedKey -> {
+                tx.delete(mergedTable, (Sample.StringKey) deletedKey);
+            });
+
+            tx.commit();
+        }
```

Comment:
you can handle the exception or print some relevant logs?

Reply:
Sure, just added a catch to handle `TransactionAbortedException` and some log messages in there and similar places.